python/tests/test_project.py

```
@@ -81,6 +81,170 @@ def test_cog_layer_restore_shape():
+    assert layer["sourcePath"] == "https://e/data.parquet"
+
+
+def test_vector_layer_tiles_mode():
+    layer = project.vector_layer("V", "https://e/data.fgb", render_mode="tiles")
+    assert layer["type"] == "vector-tiles"
+    assert layer["source"]["type"] == "vector"
+    assert layer["metadata"]["vectorState"]["renderMode"] == "tiles"
+
+
+def test_vector_layer_invalid_render_mode():
+    with pytest.raises(ValueError):
```

Ruff (0.15.15)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)